docs/enhancements/third-party-cni-support.md

```diff
@@ -0,0 +1,274 @@
+- Known Constraints: OSP 16.2 with OCP 4.15/4.16 not supported for production use
+
+**Isovalent Cilium**
+- Supported Platforms: Bare Metal, vSphere, AWS, Azure, GCP, OpenStack, Nutanix
+- Supported OCP Versions: 4.5 - 4.20
+- Cilium Versions: 1.9 - 1.18
+- Installer Types: UPI, IPI (version-dependent)
+- Known Constraints: Some virtualization test results may be pending
+
+**Tigera Calico**
+- Supported Platforms: Bare Metal, vSphere, AWS, Azure, GCP, OpenStack
+- Supported OCP Versions: 4.8 - 4.18
+- Calico Versions: 3.20 - 3.30
+- Installer Types: IPI, UPI (version-dependent)
+
+#### API Changes
+
+Extend the cluster creation API to accept CNI selection through a new `network_type` field in `ClusterCreateParams`. The field will support the following values:
+- `OVNKubernetes` (default for supported versions)
+- `OpenShiftSDN` (for older OpenShift versions where still supported)
+- `CiscoACI`
+- `Cilium`
+- `Calico`
+- `None` (no CNI installed, user must provide manifests)
+
+**CNI Version Handling:**
+Assisted Installer will not manage specific CNI versions (e.g., Cilium 1.17 vs 1.18). The CNI version is determined by the manifests provided by the user. Assisted Installer only validates that the selected CNI type is compatible with the OCP version and platform. Users obtain version-specific manifests from CNI vendors and upload them via the custom manifests API.
+
+Validation requirements:
+- Third-party CNIs require `user_managed_networking: true` (cluster-managed networking supports OVN-K and OpenShiftSDN only)
+- API validates CNI against support matrix (platform, OCP version, topology)
+- Returns HTTP 400 with descriptive error for unsupported combinations
+- `network_type: None` requires custom manifests to be uploaded via the manifests API
+- OpenShiftSDN availability is version-dependent (deprecated in newer OCP releases)
+
+#### Changes to assisted-service
+
+**Provider Abstraction:**
+Implement a CNI provider registry pattern (similar to existing platform providers) that allows each CNI to register its support matrix, validation rules, and documentation links. The registry will be consulted during cluster validation to determine CNI compatibility.
+
+**Validation Improvements:**
+Extend cluster validation logic to check CNI compatibility with the selected platform, OCP version, and cluster topology. Validation will reference a structured support matrix (stored as JSON) that maps CNI support to specific OCP versions and platforms.
+
+Add new validation that blocks installation if a third-party CNI is selected but no custom manifests have been uploaded. The validation will check for the presence of manifests and display an error message directing users to upload CNI manifests before proceeding.
+
+**Manifest Generation:**
+Modify install-config generation to skip OVN-K-specific configuration when a third-party CNI is selected. The CNI will be configured via custom manifests provided by the user through the existing manifests API.
```

Comment:
Please add the mapping between the network_type enum and the value in the install-config